### ISSUE TITLE
Does not work in SLIME and Emacs 28.1

### ISSUE THREAD

**Issue by svetlyak40wt** [gorozhin/chlorophyll#2]
Hi! I've installed [slime-repl-ansi-color][https://github.com/deadtrickster/slime-repl-ansi-color) extension, and tests how Chlorophyll works (except bold). But seems it does nothing.

Compare it's output with raw ASCII sequence and another CL library for colored output [cl-ansi-text](https://github.com/pnathan/cl-ansi-text):

<img width="714" alt="Screen Shot 2023-02-21 at 22 07 14" src="https://user-images.githubusercontent.com/24827/220436548-eacd6f47-e3ba-49e6-b60e-ca3de8b52fa9.png">

The same problem is in the SLY:

<img width="716" alt="Screen Shot 2023-02-21 at 22 04 53" src="https://user-images.githubusercontent.com/24827/220436114-1143f2a2-5b1d-4f56-be5d-5a98fdc55ce7.png">

However when I try to execute code in raw SBCL repl, it works (except that underline does not work for some reason):

<img width="651" alt="Screen Shot 2023-02-21 at 22 01 40" src="https://user-images.githubusercontent.com/24827/220435541-a08abd97-c9f6-4ab8-a06a-7821d52c1796.png">

Here is the full code to reproduce the problem:

```lisp
CL-USER> (ql:quickload :chlorophyll)
CL-USER> (ql:quickload :cl-ansi-text)

CL-USER> (format t "~c[31mabc~c[0mqweqwe~%" #\ESC #\ESC)
abcqweqwe
NIL
CL-USER> (let ((very-peri (chlorophyll:create-rgb-color 102 103 171))
      (hot-pink (chlorophyll:create-rgb-color 255 105 180)))
  (format t "~A~%~A~%" (chlorophyll:overline "overline text")
          (chlorophyll:stylize
           (chlorophyll:new-style
            :bold t
            :foreground hot-pink
            :background very-peri)
           "beautiful text")))
overline text
beautiful text
NIL
CL-USER> (cl-ansi-text:with-color (:red)
           (princ "Gets printed red...")
           (princ "and this too!"))
Gets printed red...and this too!
"and this too!"

CL-USER> (format t "~A and not bold~%"
                 (chlorophyll:bold "Bold"))
Bold and not bold

```

I'm using iTerm2 under OSX.

**Comment by gorozhin** [comment]
Hi, thanks for the attention forward the chlorophyll library.

The issue is actually a complicated one, as it includes multiple components that are out of scope for the library itself. 

chlorophyll uses 24-bit color in the example provided, cl-ansi-text, however, uses 3-bit colors by default, slime-repl-ansi-color uses emacs builltin `ansi-color-apply-on-region`, that supports 3-bit color from what I can tell by briefly testing it.

To get this behaviour using cl-ansi-text you can actually run this snippet, it uses 24-bit colors explicitly:

``` common-lisp
(let ((cl-ansi-text:*color-mode* :24bit))
  (cl-ansi-text:with-color ((cl-colors2::rgb 1.0 0 0))
    (princ "Gets printed red...")))
```

Overline is a just a rarely supported code, I guess iTerm2 does not support it.

I actually plan to add support for 3-bit colors with fallbacks (24->8->3-bit colors) depending on terminal features, but this requires a little more research.

**Comment by svetlyak40wt** [comment]
> I actually plan to add support for 3-bit colors with fallbacks (24->8->3-bit colors) depending on terminal features, but this requires a little more resesarch.

That whould be cool!

**Comment by gorozhin** [comment]
I just added the support for 3 bit colors as well as fallbacks for different color spaces (24 bit/truecolor/rgb-color -> ansi-256 colors -> 3 bit colors -> no color) based on the env variables of the terminal emulator you run. However, slime-repl-ansi-color has no means of signaling the availability of the color space it works in, so you'll need to specify it by hand, but it works.

``` common lisp
(format t (chlorophyll:stylize
           (chlorophyll:new-style
            :background (chlorophyll:create-rgb-color 255 220 4))
           "hello"
           :profile (chlorophyll:create-profile (chlorophyll:new-ansi-terminal))))
```

<img width="554" alt="image" src="https://user-images.githubusercontent.com/4953453/226132694-dc18e3e2-0255-43b8-8a6b-0f523266c121.png">

**Comment by svetlyak40wt** [comment]
It would be nice to have a dynamic variable `chlorophill:*profile*` which is used as a default value for profile argument of `chlorophill:stylize` function. This way profile can be initialized only once and user will haven't to change stylizing code to support both console and SLIME or SLY.

**Comment by gorozhin** [comment]
Added `*profile*`
